### ISSUE TITLE
Welcome, Outreachy aspirants! (Summer-2020)

### ISSUE THREAD

**Issue by ebarry** [publiclab/plots2#7617]
Hello folks from Outreachy - We realized we don't have a main place for you all to ask questions and get support, so we wanted to say WELCOME and create this issue as a discussion zone. Please leave a comment to say hello, in part so we know who's interested in the program! @publiclab/reviewers @publiclab/plots2-reviewers @publiclab/mentors please keep an eye open for opportunities to welcome and support our Outreachy applicants! 👍 🎉 

## First contribution

It's so great to see that several of you have made your first contribution. 🙌 This should really help as all our contributions follow this workflow. If you haven't yet, please see this page to get started:

https://code.publiclab.org

## What next?

If you're looking for a next project, we might recommend one which starts to involve multiple files in the application code, so you can start to get a better understanding of the application's workings. Are you familiar with how a Rails application works, in terms of how the model accesses the database, the controller performs the logic, and the views display the actual web pages? If not, we can help you start to pick apart how our system works one piece at a time. 

You can find a lot of relatively small but not completely documented issues here:

https://github.com/publiclab/plots2/labels/fto-candidate

If you choose one, please leave a comment and we can do our best to help find any missing information or help guide you through the next steps. Don't hesitate to ask questions!

Larger collections of tasks can often be found under the `break-me-up` label, which you can go through to find smaller projects to "break off" and solve:

https://github.com/publiclab/plots2/labels/break-me-up

## Supporting tasks

Another area you might be interested in contributing is in supporting other new people to the project. This is an often unrecognized part of a healthy collaborative project, so we've made these issues to encourage people to take part in this crucial activity. We'd love your help, and as a newcomer yourself, you know exactly what it feels like to be new, so you know how important it is to be welcoming and supportive!

https://github.com/publiclab/plots2/labels/support

You can help new contributors in getting started by creating new FTO issues - https://github.com/publiclab/plots2/issues/4963

### Outreachy Projects

**1. https://www.outreachy.org/outreachy-may-2020-internship-round/communities/public-lab/#fortify-public-lab-text-editor-and-introduce-more-**

GitHub issue: COMING SOON

**2. https://www.outreachy.org/outreachy-may-2020-internship-round/communities/public-lab/#optimize-performance-and-accessibility-of-public-l**

GitHub issue: COMING SOON

### CALL FOR OUTREACHY PROPOSALS

We've posted a call for proposals here, where we'd like to ask each of you to open a draft proposal and get some comments on it! Thank you!!

https://publiclab.org/notes/liz/03-09-2020/call-for-outreachy-2020-proposals

Thanks and we're happy to answer questions!!

(building on previous versions of this post at #6285, #5022 and https://publiclab.org/notes/gauravano/03-12-2019/call-for-outreachy-2019-proposals)

**Comment by Janetsomehow** [comment]
Hello. Thank you for this discussion channel.

**Comment by justinelai** [comment]
Hello from NYC! Happy to be here. Thanks for outlining these options -- it can be intimidating seeing the FTO issues getting snapped up.

**Comment by theOpetunde** [comment]
Hello. I am happy to be applying as an Outreachy intern with Public Lab.

Over the past month, I have opened three pull requests, two of which have been merged [here](https://github.com/publiclab/plots2/pull/7380) and [here](https://github.com/publiclab/plots2/pull/7521).

I believe in the system Public lab is building because it is for a cause that will make the world a better place and I am more than willing to contribute in any way that I can. 

I have experience with documentation and developing with Python and Django. Here are links to some of the works I have done:
[](https://github.com/theOpetunde/todoapp-Django)
[](https://github.com/theOpetunde/profiles-rest-api)

Please let me know what I can do to prove my relevance to your project. I hope you consider my application and I look forward to hearing from you. Thank you very much

**Comment by jywarren** [comment]
Hello all, and welcome! I'm Jeffrey Yoo Warren, joining from Providence, Rhode Island, USA, and I've been a mentor/administrator for our Outreachy program for the past few years. We just wrapped up an awesome season with @crisner and @nstjean with @sagarpreet-chadha and @cesswairimu mentoring, and are excited to be working on a new project this coming season! 

Thanks everyone for showing up and saying hello! We're also happy to answer any questions or offer advice where we can! 🎉 ❤️ 👍

**Comment by jywarren** [comment]
Regarding first-timer-only issues @justinelai we can get some queued up for you if you'd like!

**Comment by theOpetunde** [comment]
@jywarren Thank you for having us all in here. Are there some other information you'd love for us to add to our application? There is a space for us to fill in any questions the organization might have. Thank you

**Comment by jywarren** [comment]
I appreciate your noting where you've participated! That could go there if it's not already in another space. Thank you! 

And folks joining with the same Q - we also love to see when you've helped others out - even just a word of encouragement, or a review or troubleshooting together in the comments - we really value this kind of interaction with others in the community and it's much appreciated, so don't hesitate to highlight where you've been involved in such ways, even if it seems more informal! Mutual support is how our community runs! 💞

**Comment by jywarren** [comment]
Ah, fixing the link, thanks!

**Comment by Shulammite-Aso** [comment]
Hello! 
I'm happy to finally be here. 
I've known about public lab for only a few weeks now. And applying as an outreachy intern for public lab is what a way to get closer to the community. 

I'm looking to start working on the project "Fortify Public Lab Text Editor and Introduce More Custom Inserts". It's a big project and I look forward to upskilling myself as I work on it. 

Meanwhile, I'm yet to start working on any issues as I haven't found any available one to claim. Please mentors, I'ld appreciate if I could get assigned an issue that I could work on for a start. 

Meanwhile, while searching around for issues, I've been able to engage with other new contributors, and I'm happy about it.

**Comment by justinelai** [comment]
@jywarren Yep, I can take on a task as they become available. (No hurry, of course -- there's plenty to read in the meantime.) Thanks!

P.S. I'm interested in the performance/a11y project. ✌️

**Comment by Medlineme4** [comment]
Hi Public Lab! It's exciting to be here. I will be glad if given assignments to work on. I so much love this contribution channel. Thanks to Outreachy, mentors and all. I can't wait to get started.

**Comment by Medlineme4** [comment]
Great contribution medium. I don't know why I have red marks in a contribution I made on FTO issue. Where did I go wrong?

**Comment by cesswairimu** [comment]
Hi @Medlineme4, I felt a comment on the pull request. Thanks

**Comment by Srija616** [comment]
Hello everyone!
I am Srija Anand from IIIT Delhi, India. I am pursuing a B.Tech in Electronics and currently in the third year. I have experience in Python though I am new to open source. @jywarren I have been trying to fix FTO issues, however unable to claim one. It would be nice if you can guide me a bit about the tasks.

**Comment by Tlazypanda** [comment]
Hey everyone! I am an Outreachy 2020 aspirant :v: 
I have been contributing to Public Lab since January.It has been an amazing journey till now :smile: .The community has been extremely helpful and has provided guidance wherever required :v: .I have had my doubts resolved with patience :sweat_smile: Initially I was really scared to contribute but the community has really made this a comfortable and safe space for me to share my ideas and thoughts :tada: 

Public Lab has done an exceptional job in making information about important environmental issues accessible to people across the Globe :100: And I hope will continue to bring substantial change in the years ahead :+1: 

During this time I have opened 16 issues and 18 pull requests.I would like to extend my thanks to @jywarren who was one of the first people I interacted with in the community :rocket: . Special thanks to @VladimirMikulic for your guidance and mentoring :smile: @cesswairimu, @SidharthBansal , @nstjean, @ebarry for all the reviews and guidance :tada: :tada: .@emilyashley Thanks for resolving my outreachy related queries on Gitter with all the onboarding stuff going on :sweat_smile: 

For my outreachy proposal, I have chosen the topic "Optimize performance and accessibility of Publiclab content " here's the initial draft for the same -  https://docs.google.com/document/d/1EBwDkBfvtHqmGZ4G3lUlpMq0KLPwyzyzg-fuUpmbTYM/edit?usp=sharing 

All suggestions and feedback is welcome :smile:

**Comment by cesswairimu** [comment]
Hi @Tlazypanda , it has been nice having you contribute to PL...Please post the proposal on the PublicLab website, you can edit as feedback comes in and its easier for mentors to leave comments and also having them read the proposals on one platform is easier for them. I hope this makes sense. Thanks

**Comment by nstjean** [comment]
Welcome all of you to Public Lab!! I am just finished my time as an Outreachy Intern here :tada:  and am very happy to help you all navigate the process.

**Comment by Srija616** [comment]
@nstjean @cesswairimu Hey! I am been making PRs for FTOs since past few days and was looking for more issues that I can take up. Can you please assign some issues. Also, I wanted to know if it is ok to make PRs even if a task hasn't been assigned to me or anyone else during Outreachy 2020 current application?

**Comment by SidharthBansal** [comment]
I am free tomorrow as it is Sunday.
So, whosoever wants feedback on their proposal please ping me
Please note: Only proposals on the public lab website will be reviewed
Secondly we highly encourage folks to help each others with the proposals
Best of luck

**Comment by Tlazypanda** [comment]
> Hi @Tlazypanda , it has been nice having you contribute to PL...Please post the proposal on the PublicLab website, you can edit as feedback comes in and its easier for mentors to leave comments and also having them read the proposals on one platform is easier for them. I hope this makes sense. Thanks

@cesswairimu @jywarren Have uploaded on pl website and edited the link above - https://publiclab.org/notes/itm2017004/03-12-2020/outreachy-proposal-optimize-performance-and-accessibility-of-public-lab-content Can you kindly provide some feedback ? Thanks :v:

**Comment by Tlazypanda** [comment]
@Janetsomehow @Srija616 @Medlineme4 @justinelai @Shulammite-Aso Have opened some ftos if you are interested to take up those :smile: #7646 #7645 #7637 #7636 Feel free to ping in case you get stuck also regarding the proposals here are some posts that you can refer-
https://github.com/prondubuisi/accepted-gsoc-proposals
You can also refer some of the proposals from last year on Public lab website.And if you are done with the proposal, feel free to ask for a review :v:

**Comment by Janetsomehow** [comment]
@Tlazypanda thank you

**Comment by Medlineme4** [comment]
Thank you

Sent from Yahoo Mail on Android 
 
  On Sun, Mar 15, 2020 at 7:22 AM, Sneha Mishra<notifications@github.com> wrote:   


@Janetsomehow @Srija616 @Medlineme4 @justinelai @Shulammite-Aso Have opened some ftos if you are interested to take up those 😄 #7646 #7645 #7637 #7636 Feel free to ping in case you get stuck also regarding the proposals here are some posts that you can refer-
https://github.com/prondubuisi/accepted-gsoc-proposals
You can also refer some of the proposals from last year on Public lab website.And if you are done with the proposal, feel free to ask for a review ✌️

**Comment by Medlineme4** [comment]
It seems they've all been taken

Sent from Yahoo Mail on Android 
 
  On Sun, Mar 15, 2020 at 7:22 AM, Sneha Mishra<notifications@github.com> wrote:   


@Janetsomehow @Srija616 @Medlineme4 @justinelai @Shulammite-Aso Have opened some ftos if you are interested to take up those 😄 #7646 #7645 #7637 #7636 Feel free to ping in case you get stuck also regarding the proposals here are some posts that you can refer-
https://github.com/prondubuisi/accepted-gsoc-proposals
You can also refer some of the proposals from last year on Public lab website.And if you are done with the proposal, feel free to ask for a review ✌️

**Comment by Jigyasa-Kumari** [comment]
Hi, I am Jigyasa, and I am an Outreachy 2020 applicant from India. Are there any issues which I can take up?

**Comment by Tlazypanda** [comment]
@Medlineme4 @Janetsomehow @Jigyasa-Kumari @Srija616 @Shulammite-Aso Opened new issues #7654 #7653 #7655

**Comment by Alexjake660** [comment]
@nstjean @Tlazypanda please I need help wrapping my head around the proposal topic. Is it a proposal on which of the two(2) projects available for interns from the outreachy program or a proposal on any project of interest in the publiclab ecosystem?

**Comment by Shulammite-Aso** [comment]
@Alexjake660 the idea is to choose a project that you would love to work on during the internship, from the projects posted on outreachy. Then study this project and know the problem that needs to be solved well enough, then write a proposal showing how you intend to solve the problems in this project during the internship.

**Comment by Medlineme4** [comment]
If you don't mind where are the projects posted

Sent from Yahoo Mail on Android 
 
  On Mon, Mar 16, 2020 at 10:20 AM, Shulammite-Aso<notifications@github.com> wrote:   


@Alexjake660 the idea is to choose a project that you would love to work on during the internship, from the projects posted on outreachy. Then study this project and know the problem that needs to be solved well enough, then write a proposal showing how you intend to solve the problems in this project during the internship.

**Comment by Shulammite-Aso** [comment]
@Medlineme4 if you're an outreachy applicant, you should be able to see the projects listed under public lab.  But also here on this page, the projects are listed https://github.com/publiclab/plots2/issues/7617#issue-578136272

**Comment by Medlineme4** [comment]
Thank you. I saw this but wasn't sure as it read coming soon

Sent from Yahoo Mail on Android 
 
  On Mon, Mar 16, 2020 at 10:36 AM, Shulammite-Aso<notifications@github.com> wrote:   


@Medlineme4 if you're an outreachy applicant, you should be able to see the projects listed under public lab. But also here on this page, the projects are listed #7617 (comment)

**Comment by Alexjake660** [comment]
@Shulammite-Aso Thank you for the clarification.

**Comment by cesswairimu** [comment]
Hi @Srija616, yeah you can work on an issue if its unassigned, just comment and say you are working on. No need for us to assign you an issue just browse the ones we have and leave a comment if if find one that you would like to work on. Thanks

**Comment by Alexjake660** [comment]
@Shulammite-Aso Please what do I do if the system test of my build on Travis fails?

**Comment by Srija616** [comment]
> Hi @Srija616, yeah you can work on an issue if its unassigned, just comment and say you are working on. No need for us to assign you an issue just browse the ones we have and leave a comment if if find one that you would like to work on. Thanks

Ok! Sure. Will work on the issues. However, it has happened that I made a PR for an open issue but later received a comment that the issue has already been resolved.

**Comment by taysa91** [comment]
Hi @Tlazypanda 
I am an Outreachy applicant and was finding my way around Github as I am new to open source.
Interested in taking up an issue but all are snatched. Any leads.
Thanks.

**Comment by Tlazypanda** [comment]
Answering some of the queries here-
1.@Alexjake660 if your build fails there's no need to worry sometimes Travis fails due to unrelated changes ...in case of this feel free to ping @cesswairimu to restart the build :smile: 
2. @Srija616 usually when you are choosing any issue to work on check if the issue already has any linked pull requests , you can find this when you scroll down the issue description if there's already a linked pull request which is recent refrain from working on that issue , try to find another one :sweat_smile: I hope this resolves your query.
3. @taysa91 Welcome to Publiclab! :tada: I will try to create some more first timers issue as soon as possible and will ping in this thread for the same , I am traveling so would take me some time for the same :sweat_smile: till then please have a look at the git workflow for the repository and feel free to ping in case you run into any issues 

Thanks!

**Comment by sssash18** [comment]
Hey folks ,Posted new ftos #7664 , #7665, #7666

**Comment by cesswairimu** [comment]
> > Hi @Srija616, yeah you can work on an issue if its unassigned, just comment and say you are working on. No need for us to assign you an issue just browse the ones we have and leave a comment if find one that you would like to work on. Thanks
> 
> Ok! Sure. @cesswairimu Will work on the issues. However, it has happened that I made a PR for an open issue but later received a comment that the issue has already been resolved.

Sorry about that...Its always best to check if anyone is working on an issue before you start working on it to avoid that in feature. Normally a comment will be left on the issue or you can see a link to a PR

**Comment by cesswairimu** [comment]
Hello all :wave: I really love the conversations, discussion and collaborations going on here :heart: . Keep it up :tada: :tada:  Please leave a comment here if you are looking for a first-timer-issue https://github.com/publiclab/plots2/issues/7683

**Comment by Tlazypanda** [comment]
Hey everyone! Opened some new ftos #7688 #7689 #7690 #7691 .Feel free to ping me in case you have any questions :sweat_smile:

**Comment by Tlazypanda** [comment]
Answering some more common queries asked - @cesswairimu @jywarren please add if I missed anything   

1. "I have worked on an fto , what's the next step" : First of all great job on your first contribution to Public Lab :tada: Now the next step is usually taking up a fto-candidate - These are issues that have a difficulty level more than fto but have all the relevant information required to solve them. So where do I find them ? https://code.publiclab.org/ Please check this website out. It has ftos, fto candidates as well as other stale issues that you can work on :smile: 

2. "My pr is not getting merged" : There might be a couple of reasons behind this. 

- Is there already an open pr for the issue that you are working on? 
If yes there might be a possibility that the linked pr might have already been reviewed and approved. 
- Is your pull request description alright ? 
Sometimes we might forget linking the pull request to the issue. For example in the pr template we have something written as "Fixes #0000" please replace this with the issue number that you are solving for example "Fixes #7689 ".Apart from this the checklist below has to be checked for the points that you have implemented for example if you have tagged @publiclab/reviewers then put an x inside the [] to make it [x] which looks something like this on preview 

- [x] .Here is a [link](https://guides.github.com/pdfs/markdown-cheatsheet-online.pdf) that helps to understand Github Markdown syntax.  

- Has the reviewer requested some changes ? 
If the reviewer has requested some changes you will have to make those changes in your pr and again notify the reviewer to review again.

3."All the fto issues are taken" : We are constantly working to bring new beginner friendly issues for you please follow this thread #7683 for the same and comment if you want more issues :v: 

Apart from this some good practices while making a pull request- 

- Please ensure your branch name is relevant to the issue like if you are working on an issue for translation in _wiki then an appropriate branch name would be 'translation_wiki'. 

- Please ensure your commit message is relevant to the issue like for the above example an appropriate commit message would be 'rename t to translation in _wiki file'.

These practices will help the ones reviewing your code and will also ensure the commit history looks awesome :100: Looking forward to some amazing contributions everyone :tada: :dancer:

**Comment by guArroyo** [comment]
Hello everyone! I'm a 2020 Outreachy applicant, and I'm looking forward to working with everyone here as the contribution phase continues! :) Cheers.

**Comment by shreyaa-s** [comment]
Hey everyone, I'm Shreyaa Sharma, an Outreachy 2020 applicant. Looking forward to working with everyone. 
All the best!

**Comment by cesswairimu** [comment]
Hi everyone, if you have been waiting for too long for aa review please feel free to leave the link here to the PR and ping me so you can smooth your application...Also any proposals on the PL website you can leave a link so as the mentors can give feedback. 
 Welcome @guArroyo @shreyaa-s

**Comment by cesswairimu** [comment]
Thanks @Tlazypanda, The explanation is perfect thank you :heart:

**Comment by nicoleiocana** [comment]
Hey everyone! Like many, I am also a potential Outreachy intern. Competition intensifies! jk. I have worked with Plots2 since November of last year and I can only echo what has already been said: Public Lab members (@jywarren , @SidharthBansal , @cesswairimu) are the most welcoming individuals to integrate you in the process of open source web development! I found this repo when I began my web development/ programming journey and needed to practice PR's. They have been nothing but awesome and patient. For those who have yet to interact with them (and the other Public Lab members who I have not had the pleasure of interacting with), you guys are in for a real treat!

@publiclab/mentors I have a question: If I have already contributed to Plots2 prior to Outreachy's application period, do I still have to contribute to a first-timers-only issue? Or would it be more appropriate to leave these issues to individuals who have never contributed to your guys' repo?

**Comment by nicoleiocana** [comment]
> Hello everyone! I'm a 2020 Outreachy applicant, and I'm looking forward to working with everyone here as the contribution phase continues! :) I see that some issues are open, should I just comment to be assigned to them or are any off limits?

@guArroyo Underneath the Assignee column, if you see the Linked PR icon  ![linkedPR](https://user-images.githubusercontent.com/44947175/77573164-7b855080-6e8d-11ea-8b25-4101c9eb05dd.PNG), then that open issue will most likely be closed because someone already attributed the resolution. So these are off limits.

Yet, most collaborators don't mind if you ask them to be assigned the issue; they will respond back with their permission within the thread.
![grantedPermission](https://user-images.githubusercontent.com/44947175/77573468-f8182f00-6e8d-11ea-8c06-890345df36be.PNG)

**Comment by cesswairimu** [comment]
@nicoleiocana, thanks for your kind words :blush: ...Happy to have you onboard.

No you do not need to contribute to another fto. You can move on to the help-wanted or bug issues. Thanks

**Comment by cesswairimu** [comment]
🌳🌟 Hi everyone, here's something very rewarding you can do in 5 minutes to benefit the entire community: you can fill out this survey and make sure YOU are represented in our 2020 Community Snapshot! https://forms.gle/Q9BNduRGzsLrALxi7 📈 

Check out the 2019 report here: https://publiclab.org/notes/liz/04-15-2019/report-2019-software-contributors-survey. We are going to do the same "chart-n-graphs" treatment for 2020 but before we can begin, we need more than 38 people to fill out this survey! That's right, so far only 38 people have filled the 2020 survey even though we have more active contributors than ever. In 2019, 63 people filled out the survey. **Let's exceed last year's response rate! Please click here to improve our data quality: https://forms.gle/Q9BNduRGzsLrALxi7**

Thank you for all you do!
💚

_Originally posted by @ebarry in https://github.com/publiclab/plots2/issues/7711#issuecomment-604460524_

**Comment by nicoleiocana** [comment]
@cesswairimu Just to be on the safe side: if I do NOT contribute to a help-wanted or bug issue, will that contribution not count towards my Outreachy requirement? I still plan to contribute by the options mentioned at the top of this thread, but I just wanna make sure that I am not doing anything wrong.

**Comment by cesswairimu** [comment]
No that's not the case, any contribution will count. Sorry for the confusion

**Comment by nicoleiocana** [comment]
@cesswairimu Oh, no need to apologize, I should have clarified. I just didn't wanna document this contribution for my Outreachy application and somehow it would not be valid. Thanks for the quick response. I still plan to give back and support first-timers-only by creating new fto issues #4963.

**Comment by Janet33** [comment]
Hello, just saw this issue and I must say...this is very encouraging!

**Comment by ebarry** [comment]
Great to meet everyone on this thread! Please join us over at the new issue "Welcome, Outreachy/GSoC 2020 participants! Let's plan & coordinate! 🎉🙋‍♀️🙋🙋‍♂️"  #7873

**Comment by khadija-nur** [comment]
Hi @Tlazypanda can I claim this issue?
